File: ruby_small_problems/easy_4/7.rb
# takes a string of numeric chars including an optional plus or minus sign
# returns an integer
#
# dont worry about leading + or - signs or invalid chars
DIGITS = {
  '0' => 0, '1' => 1, '2' => 2, '3' => 3, '4' => 4,
  '5' => 5, '6' => 6, '7' => 7, '8' => 8, '9' => 9
}

def string_to_integer(str)
  digits = str.chars.map{ |x| DIGITS[x]  }
  value = 0
  digits.each { |x| value = (10 * value) + digit }
  value
end

File: 210/exercises/javascript_basics/7.js
const DIGITS = {
  0: 0,
  1: 1,
  2: 2,
  3: 3,
  4: 4,
  5: 5,
  6: 6,
  7: 7,
  8: 8,
  9: 9,
}

function stringToInteger(string) {
  let multiplier = 1;
  let i;
  let result = 0;
  for (i = string.length - 1; i >= 0; i -= 1) {
    result += DIGITS[string[i]] * multiplier; 
    multiplier *= 10;
  }
  return result;
}

console.log(stringToInteger('4321'));      // 4321
console.log(stringToInteger('570'));       // 570
